src/sqlancer/postgres/oracle/PostgresCERT.java

```diff
@@ -0,0 +1,197 @@
+            throw new AssertionError("Inconsistent result for query: EXPLAIN " + queryString1 + "; --" + rowCount1
+            + "\nEXPLAIN " + queryString2 + "; --" + rowCount2);
+        }
+    }
+
+    @Override
+    protected boolean mutateDistinct() {
+        PostgresSelect.SelectType selectType = select.getSelectOption();
+        if (selectType != PostgresSelect.SelectType.ALL) {
+            select.setSelectType(PostgresSelect.SelectType.ALL);
+            return true;
+        } else {
+            select.setSelectType(PostgresSelect.SelectType.DISTINCT);
+            return false;
+        }
+    }
+
+    @Override
+    protected boolean mutateWhere() {
+        boolean increase = select.getWhereClause() != null;
+        if (increase) {
+            select.setWhereClause(null);
+        } else {
+            select.setWhereClause(gen.generateExpression(0));
+        }
+        return increase;
+    }
+
+    @Override
+    protected boolean mutateGroupBy() {
+        boolean increase = select.getGroupByExpressions().size() > 0;
+        if (increase) {
+            select.clearGroupByExpressions();
+        } else {
+            select.setGroupByExpressions(select.getFetchColumns());
+        }
+        return increase;
+    }
+
+    @Override
+    protected boolean mutateHaving() {
+        if (select.getGroupByExpressions().size() == 0) {
+            select.setGroupByExpressions(select.getFetchColumns());
+            select.setHavingClause(gen.generateExpression(0));
+            return false;
+        } else {
+            if (select.getHavingClause() == null) {
+                select.setHavingClause(gen.generateExpression(0));
+                return false;
+            } else {
+                select.setHavingClause(null);
+                return true;
+            }
+        }
+    }
+
+    @Override
+    protected boolean mutateAnd() {
+        if (select.getWhereClause() == null) {
+            select.setWhereClause(gen.generateExpression(0));
+        } else {
+            PostgresExpression newWhere = new PostgresBinaryLogicalOperation(select.getWhereClause(),
+                    gen.generateExpression(0), PostgresBinaryLogicalOperation.BinaryLogicalOperator.AND);
+            select.setWhereClause(newWhere);
+        }
+        return false;
+    }
+
+    @Override
+    protected boolean mutateOr() {
+        if (select.getWhereClause() == null) {
+            select.setWhereClause(gen.generateExpression(0));
+            return false;
+        } else {
+            PostgresExpression newWhere = new PostgresBinaryLogicalOperation(select.getWhereClause(),
+                    gen.generateExpression(0), PostgresBinaryLogicalOperation.BinaryLogicalOperator.OR);
+            select.setWhereClause(newWhere);
+            return true;
+        }
+    }
+
+    private int getRow(SQLGlobalState<?, ?> globalState, String selectStr, List<String> queryPlanSequences)
```

Comment:
This function should retrieve the cardinality estimation from the query plans of PostgreSQL, and is not exactly the same the one as MySQL.

You can check here: https://www.postgresql.org/docs/current/using-explain.html to know the query plan formats, and redesign this function to extract the correct cardinality.

Reply:
Thanks! 
Was looking for this only. 
I realized it latter that Postgres implementation will be a little different than the MySQL one. Will update the PR as I fix these issues.

Reply:
I did come across some problematic queries that were being generated by the [PostgresExpressionGenerator](https://github.com/sqlancer/sqlancer/blob/main/src/sqlancer/postgres/gen/PostgresExpressionGenerator.java) with errors mostly related to default function formats(eg RPAD) and errors related to data types.
Should I follow a similar approach that was taken in MySQL by adding the expected errors in  [MySQLErrors.java](https://github.com/sqlancer/sqlancer/blob/main/src/sqlancer/mysql/MySQLErrors.java)?